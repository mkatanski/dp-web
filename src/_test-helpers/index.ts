export * from "./withResponsiveContext";
export * from "./withThemeProvider";
