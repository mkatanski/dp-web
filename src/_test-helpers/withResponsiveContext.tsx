import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
