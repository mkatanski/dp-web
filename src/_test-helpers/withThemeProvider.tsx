import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
