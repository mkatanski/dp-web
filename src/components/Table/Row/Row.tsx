import styled from "styled-components";
import { TableRow } from "@material-ui/core";

export const Row = styled(TableRow)`
  cursor: ${({ onClick }) => (onClick ? "pointer" : "default")};
`;
