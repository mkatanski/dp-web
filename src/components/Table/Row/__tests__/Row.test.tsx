import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

/* Uncomment below if needed */
import { withThemeProvider } from "_test-helpers";
import { render, fireEvent } from "@testing-library/react";

import { Row } from "../Row";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

/* Component mock example */
// jest.mock("[module]", () => ({
//   [ComponentName]: ({children, ...rest}: React.PropsWithChildren<{}>) => (
//     <div id="[ComponentName]" {...rest}>{children}</div>
//   )
// }));

describe("components", () => {
  describe("Row", () => {
    it("should render Row component", () => {
      const tree = TestRenderer.create(withThemeProvider(<Row />));

      expect(tree).toMatchSnapshot();
    });

    it("should render Row component with onClick", () => {
      const tree = TestRenderer.create(
        withThemeProvider(<Row onClick={() => null} />)
      );

      expect(tree).toMatchSnapshot();
    });

    it("should handle onClick event", async () => {
      const handleClick = jest.fn();
      const tableRow = document.createElement("tbody");
      const { findByText } = render(
        withThemeProvider(
          <Row onClick={handleClick}>
            <td>Test</td>
          </Row>
        ),
        {
          container: document.body.appendChild(tableRow)
        }
      );
      fireEvent.click(await findByText("Test"));

      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
