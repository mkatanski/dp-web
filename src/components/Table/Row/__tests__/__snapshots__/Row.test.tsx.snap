// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Row should render Row component 1`] = `
.c0 {
  cursor: default;
}

<tr
  className="c0"
  id="MUI.TableRow"
/>
`;

exports[`components Row should render Row component with onClick 1`] = `
.c0 {
  cursor: pointer;
}

<tr
  className="c0"
  id="MUI.TableRow"
  onClick={[Function]}
/>
`;
